card/lib/card.rb

```diff
@@ -53,8 +53,9 @@ def self.serializable_attr_accessor *args
   attr_accessor :follower_stash
 
 
-  define_callbacks :approve, :store, :stored, :extend, :subsequent
+  define_callbacks :prepare, :approve, :store, :stored, :extend, :subsequent, :select_action
 
+  before_validation :prepare
```

Comment:
I had sort of been leaning this way a bit.  Kind of feels like we need to put all the events on post-its and arrange and re-arrange until everything makes sense...